src/core_plugins/metrics/public/components/vis_editor_visualization.js

```diff
@@ -62,8 +64,35 @@ class VisEditorVisualization extends Component {
   componentWillUnmount() {
     window.removeEventListener('mousemove', this.handleMouseMove);
     window.removeEventListener('mouseup', this.handleMouseUp);
+    if (this._handler) {
+      this._handler.destroy();
+    }
+  }
+
+  _loadVisualization() {
+    getVisualizeLoader().then(loader => {
+      if (!this._visEl.current) {
+        return;
```

Comment:
Let's add a small comment like `In case the visualize loader isn't done before the component is unmounted.` just to make a bit clearer for what we have this drop out here.